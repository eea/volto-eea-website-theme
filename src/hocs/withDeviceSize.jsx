import React from 'react';

export default function withDeviceSize(WrappedComponent) {
  return (props) => {
    const [device, setDevice] = React.useState(null);

    const updateScreenSize = () => {
      if (__CLIENT__) {
        const screenWidth =
          document.documentElement.clientWidth ||
          document.body.clientWidth ||
          window.innerWidth ||
          0;

        setDevice(getDeviceConfig(screenWidth));
      }
    };

    const getDeviceConfig = (width) => {
      // semantic ui breakpoints
      if (width < 768) {
        return 'mobile';
      } else if (width >= 768 && width < 992) {
        return 'tablet';
      } else if (width >= 992 && width < 1200) {
        return 'computer';
      } else if (width >= 1200 && width < 1920) {
        return 'large';
      } else if (width >= 1920) {
        return 'widescreen';
      }
    };

    React.useEffect(() => {
      updateScreenSize();
      window.addEventListener('resize', updateScreenSize);
      return () => {
        window.removeEventListener('resize', updateScreenSize);
      };
      /* eslint-disable-next-line */
    }, []);

    return <WrappedComponent {...props} device={device} />;
  };
}
