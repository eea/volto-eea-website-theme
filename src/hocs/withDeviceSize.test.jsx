import React from 'react';
import withDeviceSize from './withDeviceSize';

import { render } from '@testing-library/react';

describe('withDeviceSize', () => {
  const Block = (props) => {
    return (
      <>
        <div>My Block</div>
      </>
    );
  };
  it('calculates size based on browser widths', () => {
    const ExtendedBlock = withDeviceSize(Block);
    const data = { '@type': 'testBlock' };
    const { container } = render(<ExtendedBlock data={data} />);
    expect(container).toMatchSnapshot();
  });
});
