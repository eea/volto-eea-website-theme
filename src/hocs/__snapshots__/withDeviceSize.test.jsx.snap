// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withDeviceSize calculates size based on browser widths 1`] = `
<div>
  <div>
    My Block
  </div>
</div>
`;
